### PR TITLE
Only include debug info for binary deployed to TileDB Cloud

### PR DESCRIPTION
**Pull request by jdblischak**
**Background:** 

* The debug info was included in the binary starting in 0.6.0 (https://github.com/TileDB-Inc/TileDB-Vector-Search/pull/424). My understanding is that the motivation for this was for easier debugging in TileDB Cloud
* Including the debug info greatly increases the size of the binary, and thus the conda binary uploaded to anaconda.org
* We are running out of space on anaconda.org

**Proposal:**

I have attempted to use the [scikit-build-core configuration](https://scikit-build-core.readthedocs.io/en/latest/configuration.html) to override the settings in `pyproject.toml`. The goal is to only include the debug info for the binary that is installed for TileDB Cloud (linux-64 and python 3.9).

**Status:**

Unfortunately it isn't currently working. Despite passing `-C skbuild.cmake.build_type="Release" ` and `--config-settings=install.strip=true `, CMake still reports `-- Install configuration: "RelWithDebInfo"`, and the binary `_tiledbvspy.cpython-310-x86_64-linux-gnu.so` is still huge.

Can anyone see how I need to edit the `pip install` command to properly override the default settings?

### CODE REVIEW

**Comment by jdblischak** [comment]
As expected from the builds on my fork, the [linux-64 python 3.9 build](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=40562&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=937c195f-508d-5135-dc9f-d4b5730df0f7) used the correct `script:` command, built the binary in "RelWithDebInfo" mode, and created a large binary. The other builds, eg [linux-64 python 3.10](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=40562&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7), used the correct `script:` with `-C skbuild.cmake.build_type="Release"` and `--config-settings=install.strip=true`, but still built the binary in "RelWithDebInfo" mode, and created a large binary.

In other words, the jinja2 preprocessing selectors are working as expected, but the command-line options I am passing to `pip install` are not having any effect on scikit-build-core.

@dudoslav any ideas?

**Comment by dudoslav** [comment]
@jdblischak Looking at the scikit-build-core documentation it seems that the correct option should be:
```
pip install . --config-settings=cmake.build-type="Debug"
```
https://scikit-build-core.readthedocs.io/en/latest/configuration.html#configuring-cmake-arguments-and-defines

So perhaps I gave you an incorrect option to set, sorry for that.

**Comment by jdblischak** [comment]
@dudoslav thanks! Trying that now

**Comment by jdblischak** [comment]
Success! The binary for linux-64 python 3.10 was reduced from 123,872,008 to 4,978,832

**Comment by jdblischak** [comment]
@NikolaosPapailiou @ihnorton is it ok if only the conda binary that is deployed to TileDB Cloud includes the debug info? This will drastically reduce the storage requirements of future releases of tiledb-vector-search on anaconda.org

**Review by NikolaosPapailiou — APPROVED** [review]

**Comment by jdblischak** [comment]
While not as drastic of a size decrease as the linux-64 binaries, the size of the osx-64 binary `_tiledbvspy.cpython-39-darwin.so` was still halved from 9,135,816 in the [most recent osx-64 python 3.9 build on main](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=40439&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=f047df50-a233-5ae9-bf8f-c7dd8e5684ad) to 4,498,128 in the [osx-64 python 3.9 build for this PR](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=40566&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=f047df50-a233-5ae9-bf8f-c7dd8e5684ad)
